c++/CMakeLists.txt

```diff
@@ -1,6 +1,7 @@
 project("Cap'n Proto" CXX)
 cmake_minimum_required(VERSION 3.1)
-set(VERSION 0.7-dev)
+set(RELEASE_VERSION 0.7)
```

Comment:
By separating the "0.7" and "-dev" parts here you're actually breaking our release scripts which automatically update this for each release.

It looks like you're doing this so that only "0.7" goes into the SONAME (library version), but that's actually *not* equivalent to what autotools does. The autotools build uses the whole "0.7-dev" as the version string to embed in the SONAME.

Note that, again, our release scripts remove the -dev automatically in the release branch, so only builds from master would ever end up with the -dev tag. (There is also the -rc1, -rc2, etc. tags that designate release candidates, but final releases have no such suffix.)

So can we revert this part of the change?